daemon/oci_linux.go

```diff
@@ -768,6 +768,14 @@ func WithCommonOptions(daemon *Daemon, c *container.Container) coci.SpecOpts {
 		s.Hostname = c.Config.Hostname
 		setLinuxDomainname(c, s)
 
+		// add default sysctls that are generally safe and useful; currently we
+		// grant the capabilities to allow these anyway. You can override if
+		// you want to restore the original behaviour.
+		// allow unprivileged ICMP echo sockets without CAP_NET_RAW
+		s.Linux.Sysctl["net.ipv4.ping_group_range"] = "0 2147483647"
+		// allow opening any port less than 1024 without CAP_NET_BIND_SERVICE
+		s.Linux.Sysctl["net.ipv4.ip_unprivileged_port_start"] = "0"
```

Comment:
Are custom values already set at this point? If so, should we skip these if a custom value was set? (not sure if someone would have any reason to)

Reply:
@yosifkit looking over my shoulder while I'm cheering for this glorious PR also asked if one or both of these should be specifically excluded when using `--network=host`? (it's not clear from the code if this is already the case :see_no_evil:)

Reply:
Good point! 👍 (not sure what it does when manually specifying `--sysctl` 🤔

Reply:
Looks like it "works" just fine? :sweat_smile:
```console
$ docker run --init -it --rm --network host --sysctl net.ipv4.ip_unprivileged_port_start=0 --user 1000:1000 tianon/network-toolbox nc -vvv -l -p 123
Listening on [0.0.0.0] (family 2, port 123)
```

Reply:
... and persistently adjusts the value for my host: :partying_face: 

```console
$ cat /proc/sys/net/ipv4/ip_unprivileged_port_start
0
```

Reply:
did it change your host's configuration? 😬

Reply:
LOL; race condition in comments.


Looks like that's a .... "feature"?

Reply:
Ah yes, better exclude net=host too...

Reply:
As a follow-up on this thread, @kolyshkin separately fixed the "`--sysctl=net.*` + `--net=host` changes the host's settings" part of this discussion in `runc` itself via https://github.com/opencontainers/runc/pull/2706 (issue https://github.com/opencontainers/runc/issues/2693). :smile: :tada: